service/history/shardController.go

```diff
@@ -42,24 +41,19 @@ const (
 
 type (
 	shardController struct {
-		service             service.Service
-		host                *membership.HostInfo
-		hServiceResolver    membership.ServiceResolver
-		membershipUpdateCh  chan *membership.ChangedEvent
-		shardMgr            persistence.ShardManager
-		historyV2Mgr        persistence.HistoryManager
-		executionMgrFactory persistence.ExecutionManagerFactory
-		domainCache         cache.DomainCache
-		engineFactory       EngineFactory
-		shardClosedCh       chan int
-		isStarted           int32
-		isStopped           int32
-		shutdownWG          sync.WaitGroup
-		shutdownCh          chan struct{}
-		logger              log.Logger
-		throttledLoggger    log.Logger
-		config              *Config
-		metricsClient       metrics.Client
+		resource.Resource
+
+		membershipUpdateCh chan *membership.ChangedEvent
+		engineFactory      EngineFactory
```

Comment:
could we get rid of the engine factory?

Reply:
history engine should be created by shard controller

Reply:
I will have another PR for refactoring shard controller, shard item, shard context and history engine construction. This will be done as part of that PR.

Reply:
Actually I think history engine should be created by shard context?

Reply:
yes engine and shard context are the used together and should be coupled, in my opinion